dev/tools/gen_localizations.dart

```diff
@@ -107,19 +107,74 @@ String generateString(String s) {
   return output.toString();
 }
 
+/// Simple data class to hold parsed locale. Does not promise validity of any data.
+class Locale {
+  Locale({
+    this.languageCode,
+    this.scriptCode,
+    this.countryCode,
+    this.length,
+    this.origString
+  });
+
+  String languageCode;
+  String scriptCode;
+  String countryCode;
+  int length;          // The number of fields. Ranges from 1-3.
+  String origString;   // Original un-parsed locale string.
+}
```

Comment:
Maybe this class should have a toString() method that does what `describeLocale` does? It seems odd that we're parsing locale strings here and in describeLocale().

Reply:
Describe locale pulls data from the web to create a mapping between codes and human-readable proper nouns. The web request stuff probably doesn't belong in this class. `describeLocale` is not a simple parse function.

Reply:
I was referring to this describeLocale method: https://github.com/flutter/flutter/blob/bac667e0f92b0afbd6f85c6f587290878cf5b6e3/dev/tools/localizations_utils.dart#L191

Reply:
Yes, `_languages`, `_region`, and `_scripts` are maps that are generated from HTTP calls in the `precacheLanguageAndRegionTags()` function above.

Reply:
Sorry, one of the risks of just looking at diffs :-)